### PR TITLE
chore: use testify instead of testing.Fatal or testing.Error in pkg

### PR DESCRIPTION
**Pull request by mmorel-35**
### Description

* uses testify instead of testing.Fatal or testing.Error in pkg

### CODE REVIEW

File: pkg/apis/application/v1alpha1/types_test.go [hunk 1]
```diff
@@ -1090,9 +1090,8 @@ func TestAppProjectSpec_DestinationClusters(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			d := AppProjectSpec{Destinations: tt.destinations}
-			if got := d.DestinationClusters(); !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("AppProjectSpec.DestinationClusters() = %v, want %v", got, tt.want)
-			}
+			got := d.DestinationClusters()
+			assert.Truef(t, reflect.DeepEqual(got, tt.want), "AppProjectSpec.DestinationClusters() = %v, want %v", got, tt.want)
```
Comment (andrii-korotkov-verkada) [inline pkg/apis/application/v1alpha1/types_test.go:1094]: This should use require. Also, let's have a second argument assigned on a separate line for better readability.

File: pkg/apis/application/v1alpha1/types_test.go [hunk 2]
```diff
@@ -1090,9 +1090,9 @@ func TestAppProjectSpec_DestinationClusters(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			d := AppProjectSpec{Destinations: tt.destinations}
-			if got := d.DestinationClusters(); !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("AppProjectSpec.DestinationClusters() = %v, want %v", got, tt.want)
-			}
+			got := d.DestinationClusters()
+			ok := reflect.DeepEqual(got, tt.want)
+			assert.Truef(t, ok, "AppProjectSpec.DestinationClusters() = %v, want %v", got, tt.want)
```
Comment (andrii-korotkov-verkada) [inline pkg/apis/application/v1alpha1/types_test.go:1095]: Should we use require?

File: pkg/apis/application/v1alpha1/types_test.go [hunk 3]
```diff
@@ -1130,9 +1130,9 @@ func TestAppProjectSpec_DestinationClusters(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			d := AppProjectSpec{Destinations: tt.destinations}
-			if got := d.DestinationClusters(); !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("AppProjectSpec.DestinationClusters() = %v, want %v", got, tt.want)
-			}
+			got := d.DestinationClusters()
+			ok := reflect.DeepEqual(got, tt.want)
+			require.Truef(t, ok, "AppProjectSpec.DestinationClusters() = %v, want %v", got, tt.want)
```
Comment (crenshaw-dev) [inline pkg/apis/application/v1alpha1/types_test.go:1135]: I think `assert.Equal` would actually work fine here. We're just comparing string slices.

File: pkg/apis/application/v1alpha1/types_test.go [hunk 4]
```diff
@@ -1435,12 +1425,8 @@ func TestResourceResults_Find(t *testing.T) {
...
 		t.Run(tt.name, func(t *testing.T) {
 			got, got1 := tt.r.Find(tt.args.group, tt.args.kind, tt.args.namespace, tt.args.name, tt.args.phase)
-			if got != tt.want {
-				t.Errorf("ResourceResults.Find() got = %v, want %v", got, tt.want)
-			}
-			if !reflect.DeepEqual(got1, tt.want1) {
-				t.Errorf("ResourceResults.Find() got1 = %v, want %v", got1, tt.want1)
-			}
+			assert.Equalf(t, tt.want, got, "ResourceResults.Find() got = %v, want %v", got, tt.want)
+			assert.Truef(t, reflect.DeepEqual(got1, tt.want1), "ResourceResults.Find() got1 = %v, want %v", got1, tt.want1)
```
Comment (crenshaw-dev) [inline pkg/apis/application/v1alpha1/types_test.go:1429]: +1, I think we can simplify. Just need a `require.NotNil` and then an `assert.Equal`.

**Review by crenshaw-dev — APPROVED** [review]
Thanks!

**Review by pasha-codefresh — APPROVED** [review]

**Pull request merged**